File: Sortings/01_Bubble_Sort/bubble_sort.py
def bubble_sort(arr:list):
    ln = len(arr)
    for i in range(ln):
        isSwapped = False
        for j in range(0, ln - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                isSwapped = True
        if not isSwapped:
            break

a = [4, 3, 2, 1]

print('Array before sorting', a)

bubble_sort(a)

print('Array after sorting', a)

File: Sortings/01_Bubble_Sort/bubble_sort.go
package main

import "fmt"

func BubbleSort(array []int) []int {
	ln := len(array)
	for i := 0; i < ln; i++ {
		var isSwapped bool = false
		for j := 0; j < ln-i-1; j++ {
			if array[j] > array[j+1] {
				isSwapped = true
				array[j], array[j+1] = array[j+1], array[j]
			}
		}
		if !isSwapped {
			break
		}
	}
	return array
}

func main() {
	a := []int{4, 3, 2, 1, 1}

	fmt.Println("Array before sotring", a)

	a = BubbleSort(a)

	fmt.Println("Array after sotring", a)
}
